From da8029438ea8f00a154cb6a1bbc8243de39690de Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sun, 18 Dec 2022 22:11:58 +0000
Subject: [PATCH] GCC: remove redundant template parameter from SyncIterator

GCC 12 in C++20 mode does not allow this anymore.

Bug: 819294
Change-Id: I6b22d3919f08fb72a5b1d69eea59c6e95259c16f
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4111547
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Reviewed-by: Kentaro Hara <haraken@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1084808}
---

diff --git a/third_party/blink/renderer/bindings/scripts/bind_gen/sync_iterator.py b/third_party/blink/renderer/bindings/scripts/bind_gen/sync_iterator.py
index 1d67fa2..9cec5094 100644
--- a/third_party/blink/renderer/bindings/scripts/bind_gen/sync_iterator.py
+++ b/third_party/blink/renderer/bindings/scripts/bind_gen/sync_iterator.py
@@ -36,7 +36,7 @@
                        member_initializer_list=[
                            "${base_class_name}(source, kind)",
                        ]),
-        CxxFuncDeclNode(name="~${class_name}",
+        CxxFuncDeclNode(name="~SyncIterator",
                         arg_decls=[],
                         return_type="",
                         override=True,
